internal/interpreter/api.rs

```diff
@@ -654,10 +656,37 @@ impl ComponentCompiler {
         &mut self,
         source_code: String,
         path: PathBuf,
+    ) -> Option<ComponentDefinition> {
+        self.build_from_versioned_source(source_code, path, None).await
+    }
+
+    /// Compile some .slint code into a ComponentDefinition
+    ///
+    /// The `path` argument will be used for diagnostics and to compute relative
+    /// paths while importing, and the version of the `SourceFileInner` will be
+    /// set to `version`
+    ///
+    /// Any diagnostics produced during the compilation, such as warnings or errors, are collected
+    /// in this ComponentCompiler and can be retrieved after the call using the [`Self::diagnostics()`]
+    /// function. The [`print_diagnostics`] function can be used to display the diagnostics
+    /// to the users.
+    ///
+    /// Diagnostics from previous calls are cleared when calling this function.
+    ///
+    /// This function is `async` but in practice, this is only asynchronous if
+    /// [`Self::set_file_loader`] is set and its future is actually asynchronous.
+    /// If that is not used, then it is fine to use a very simple executor, such as the one
+    /// provided by the `spin_on` crate
+    pub async fn build_from_versioned_source(
```

Comment:
Can we we make this function `#[doc(hidden)]`  and `#[cfg(feature = "internal")]` ?
I don't think it shold be in our public API at this point.

Reply:
Wouldn’t we be on the safer side with a cfg in addition? Or is there a good reason to allow users to call this?

Reply:
you can add a non-`pub` version that is called bu the unversioned one.